sirocco-rt/sirocco
# Fix how we deal with extra diagnostics

I've rewritten get_extra_diagnostics and added some functions to rdpar. The basic idea of this is the following.

If you want to operate extra diagnostics - for example, getting in cell photon statistics or adjusting the grid scale lengths - then you need to run python with the following command for the cv_standard model

```
py -d cv_standard
```

You then specify the extra diagnostics in a file called ```cv_standard.extra``` which python looks for on start up. If it isn't there, it doesn't worry about extra diagnostics. If it is, it reads it into a structure just like the normal input file

Then, each time it gets to a point in the code where it needs to ask an advanced question, it looks in this file for the relevant entry. It doesn't care whether things are in a certain order but does check for duplication.  

**To test / use:**

I've done this in a new branch called extra_diags. Checkout that branch and try out the code - the example files in $PYTHON/examples/core/ should be up to date with the code. You can do this with something like:

```
git fetch upstream
# This will fetch all of the remote branches for you. 
# You can see the branches available for checkout with:
git branch -v -a

# With the remote branches in hand, you now need to check out the branch you are interested in, 
# giving you a local working copy:

git checkout -b extra_diags upstream/extra_diags
```
